back-nest/src/getAllElement/products.services.ts

```diff
@@ -0,0 +1,36 @@
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Product } from '../models/product.enity.model';
+import { RedisService } from '../redis/redis.service';
+import { CacheKeys } from '../redis/cache-keys.constant';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<Product>,
+    private readonly redisService: RedisService,
+  ) {}
+
+  async getAllProducts(): Promise<Product[]> {
+    const cachedProducts = await this.redisService.get<Product[]>(
+      CacheKeys.PRODUCTS,
+    );
+    if (cachedProducts) {
+      Logger.log('Returning products from cache');
```

Comment:
Why do you use `Logger` as a straight class here, but in the other service, it's being used as `this.logger`?

I always recommend to initialize logger as a class attribute

```ts
class MyClass {
  private readonly logger = new Logger(MyClass.name)


 async myMethod() {
   this.logger.info('')
  }
}
```